### ISSUE TITLE
Better handling around manifest directories

### ISSUE THREAD

**Issue by Michael-F-Bryan** [lu-zero/cargo-c#25]
I just tried to call `cargo cbuild` from my `examples/` directory and it looks like `cbuild` blindly adds `Cargo.toml` to the current directory when looking for the manifest.

```console
$ cargo cbuild --features std
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error(CargoMetadata("error: manifest path `/home/michael/Documents/gcode-rs/gcode/examples/Cargo.toml` does not exist\n"), State { next_error: None, backtrace: InternalBacktrace { backtrace: None } })', src/libcore/result.rs:997:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

Would it be possible to add a `--manifest-path` argument to the program and extend the manifest resolution function so it will recursively check parent directories until it finds a `Cargo.toml` or reaches `/`? I believe this is the way `cargo` works, so it'd be nice to stay consistent.

**Comment by lu-zero** [comment]
Definitely it should be addressed and probably we should document what we'll have do duplicate so we could help on the cargo-as-library-crate effort with some usecase :)

Thank you for spotting it.

**Comment by lu-zero** [comment]
Fixed in 0.6.0
